vimeo/VimeoNetworking
# VIM-7331: Move Google SignIn Helpers into Shared Library

#### Ticket

[VIM-7331](https://vimean.atlassian.net/browse/VIM-7331)

#### Pull Request Checklist


#### Issue Summary

- Adds Google Sign in API (moved from iOS client)
- Breaking changes for client on related public interfaces

#### Implementation Summary

- `AuthenticationController`: Adds authentication methods for Google Login and Join. Two changes from previous implementation 
   - we now inject the scopes 
   - the `marketingOptIn` parameter across all authentication methods is always passed in as a Bool, and then converted to a String (the expected format by the Vimeo API) inside of the request builder 
- `Constants`: Added an enum `RawTypeConverter` which intentionally wraps the conversion of any raw types to those expected by the API, such as Bool to String "false" or "true". While we could easily do this in line using `String(aBool)` or `aBool.description`, because bad request parameter serialization can be hard to diagnose, I preferred that there would be a single point of conversion that 
    - (1) does not rely on the fact that the expected type just so happens to match how Swift converts `Bool` to `String` and 
    - (2) can easily be updated if requirements change
- `RawTypeConverterTests`: tests the above
- `Request+Authentication`: 
   - Adds the request builders for Google Login and Join. 
   - Converts `marketingOptIn` to String before adding them into the request parameter dictionary

#### How to Test

- Confirm unit tests pass
- See client PR for testing that authentication flow still works
